app/questionnaire/path_finder.py

```diff
@@ -251,13 +254,15 @@ def evaluate_skip_conditions(
 
         return when_rule_evaluator.evaluate(skip_conditions["when"])
 
-    def _get_next_block_id(self, rule):
+    def _get_next_block_id(self, rule: Mapping) -> str:
         if "group" in rule:
-            return self.schema.get_first_block_id_for_group(rule["group"])
-        return rule["block"]
+            # by this point the block for the rule will exist
+            return self.schema.get_first_block_id_for_group(rule["group"])  # type: ignore
+        # the rules block will be a string
+        return rule["block"]  # type: ignore
 
     def _remove_current_blocks_answers_for_backwards_routing(
-        self, rule: dict, this_location: Location
+        self, rule: Mapping, this_location: Location
```

Comment:
From our type hints doc: `Specify at least the top-level type parameters for all generic types`.

Reply:
added